lib/ASN1/Universal/Integer.php

```diff
@@ -42,15 +42,20 @@ public function getContent()
         return $this->value;
     }
 
+    public function rightShift($number, $positions)
+    {
+        // Shift 1 right = div / 2
+        return gmp_strval(gmp_div($number, gmp_pow(2, (int)$positions)));
+    }
+
     protected function calculateContentLength()
     {
         $nrOfOctets = 1; // we need at least one octet
-        $tmpValue = abs($this->value);
-        while ($tmpValue > 127) {
-            $tmpValue = $tmpValue >> 8;
+        $tmpValue = gmp_abs(gmp_init($this->value));
```

Comment:
Hmm good point @aztech-dev. I was worried about maybe dropping the sign, but it could end up being worse than that altogether - I will add the base and leave gmp_init in to be safe! 

One of the latest/scariest in a line of related problems: http://www.darkreading.com/vulnerabilities---threats/php-hash-comparison-weakness-a-threat-to-websites-researcher-says-/d/d-id/1320353